ouster_client/CMakeLists.txt

```diff
@@ -41,6 +42,19 @@ target_include_directories(ouster_client SYSTEM PUBLIC
   $<INSTALL_INTERFACE:include/optional-lite>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/optional-lite>)
 
+# ==== Export header ====
+
+set(ouster_client_export_filename
+  ouster_client/ouster/ouster_client_export.h)
+
+generate_export_header(ouster_client 
```

Comment:
Generating `ouster_client_export.h` here, on CMAKE build, using a standard CMAKE approach.
Then, I'm using `OUSTER_CLIENT_EXPORT` in code to export the symbols, a macro is defined in the generated header, marking functions, classes, etc.
During a Windows build this basically translates to __declspec(__dllexport). With this macro one can fine-tune which symbols to export.

For ouster_osf, ouster_pcap and ouster_viz It seemed safe to just go with exporting all symbols, see this line as an example:
https://github.com/ouster-lidar/ouster-sdk/blob/5df706c7d241c7182a1b14283262da0d09072be3/ouster_osf/CMakeLists.txt#L150